### PR TITLE
go rewrite - some compute and AD diffs

### PR DESCRIPTION
**Pull request by c2thorn**
lots of small improvements from testing with AD and compute products. Still plenty of diffs from whitespace + custom code not yet implemented

does not add the compute go_yaml yet

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/products/activedirectory/Domain.yaml [hunk 1]
```diff
@@ -74,8 +74,8 @@ parameters:
     url_param_only: true
     immutable: true
     description: |
-      The fully qualified domain name. e.g. mydomain.myorganization.com, with the restrictions,
-      https://cloud.google.com/managed-microsoft-ad/reference/rest/v1/projects.locations.global.domains.
+      The fully qualified domain name. e.g. mydomain.myorganization.com, with the restrictions
+      of https://cloud.google.com/managed-microsoft-ad/reference/rest/v1/projects.locations.global.domains.
```
Comment (c2thorn) [inline mmv1/products/activedirectory/Domain.yaml:78]: the previous description was too hard to differentiate in the `description-copy.go` script, so I just slightly modified it

File: mmv1/templates/terraform/examples/base_configs/iam_test_file.go.tmpl [hunk 2]
```diff
@@ -233,13 +232,13 @@ func TestAcc{{ $.ResourceName }}IamBindingGenerated_withAndWithoutCondition(t *t
...
 				ResourceName:      "{{ $.IamTerraformName }}_binding.foo2",
-				ImportStateId:     fmt.Sprintf("{{ $.IamResourceUriFormat }} {{ $.IamPolicy.AllowedIamRole }}", {{ $.IamImportQualifiersForTest }}, {{ $example.PrimaryResourceName }}, context["condition_title"]),
+				ImportStateId:     fmt.Sprintf("{{ $.IamResourceUriFormat }} {{ $.IamPolicy.AllowedIamRole }} %s", {{ $.IamImportQualifiersForTest }}, {{ $example.PrimaryResourceName }}, context["condition_title"]),
 				ImportState:       true,
 				ImportStateVerify: true,
 			},
 			{
 				ResourceName:      "{{ $.IamTerraformName }}_binding.foo3",
-				ImportStateId:     fmt.Sprintf("{{ $.IamResourceUriFormat }} {{ $.IamPolicy.AllowedIamRole }}", {{ $.IamImportQualifiersForTest }}, {{ $example.PrimaryResourceName }}, context["condition_title_no_desc"]),
+				ImportStateId:     fmt.Sprintf("{{ $.IamResourceUriFormat }} {{ $.IamPolicy.AllowedIamRole }} %s", {{ $.IamImportQualifiersForTest }}, {{ $example.PrimaryResourceName }}, context["condition_title_no_desc"]),
```
Comment (zli82016) [inline mmv1/templates/terraform/examples/base_configs/iam_test_file.go.tmpl:241]: Good catch.

File: mmv1/templates/terraform/env_var_context.go.tmpl [hunk 3]
```diff
@@ -1,32 +1,32 @@
 {{- define "EnvVarContext" }}
 	{{- range $varKey, $varVal := $.TestEnvVars }}
-		{{- if eq $varVal $.ORGID }}
+		{{- if eq $varVal (index $.TestEnvVars "ORG_ID") }}
```
[the remark this answers is not in the document]
Comment (c2thorn) [reply mmv1/templates/terraform/env_var_context.go.tmpl:3]: They get overwritten: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/api/resource/examples.go#L184

File: mmv1/templates/terraform/examples/base_configs/iam_test_file.go.tmpl (hunk 2 above)
Comment (c2thorn) [reply mmv1/templates/terraform/examples/base_configs/iam_test_file.go.tmpl:241]: lots of little things like this. Tedious work!

File: mmv1/templates/terraform/env_var_context.go.tmpl (hunk 3 above)
[the remark this answers is not in the document]
Comment (c2thorn) [reply mmv1/templates/terraform/env_var_context.go.tmpl:3]: Let me double check

[the remark this answers is not in the document]
Comment (c2thorn) [reply mmv1/templates/terraform/env_var_context.go.tmpl]: You are correct. There was a bug in the test TestEnvVars, please see the commit I just pushed

File: mmv1/api/resource/examples.go [hunk 4]
```diff
@@ -179,8 +179,8 @@ func (e *Examples) UnmarshalYAML(n *yaml.Node) error {
 
 // Executes example templates for documentation and tests
 func (e *Examples) SetHCLText() {
-	docCopy := e
-	testCopy := e
+	docVars := make(map[string]string)
+	testVars := e.TestEnvVars
```
Comment (zli82016) [inline mmv1/api/resource/examples.go:183]: This line and line 206 are not needed?

Comment (c2thorn) [reply mmv1/api/resource/examples.go:183]: documentation and tests need to have different `TestEnvVars`. Previously, I had 
```
	docCopy := e
	testCopy := e

Comment (c2thorn) [reply mmv1/api/resource/examples.go:183]: 183 keeps a reference of the original `TestEnvVars`, 203 temporarily switches to the new vars merged with `doc_defaults`, and then 206 switches it back.

Comment (zli82016) [reply mmv1/api/resource/examples.go:183]: Got it. Thanks.

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**